File: C2_-_Input_processing_and_output/E1_-_Saying_hello/CPP/main.cpp


#include <iostream>

#define MAX_BUFFER_SIZE ( ( size_t )80 )

int main( void )
{
    char buffer[ MAX_BUFFER_SIZE ];

    std::cout << "What is your name? ";
    std::cin >> buffer;
    std::cout << "Hello " << &buffer[ 0 ] << ", nice to meet you!" << std::endl;

    return 0;
}

File: C2_-_Input_processing_and_output/E1_-_Saying_hello/RUST/main.rs
use std::io::stdin;

fn main()
{
    println!( "Hello, What is your name?" );

    let mut buffer = String::new();

    let _ret = stdin().read_line( &mut buffer );


    let res = match buffer.trim_end()
    {
        name => format!( "Hello, {name}. Nice to meet you!"),
    };


    println!("{}", res);
}